### PR TITLE
Add docs and basic integration test

### PR DESCRIPTION
**Pull request by rossf7**
Adds docs for the library and a first basic integration test.

This is an example test and also tests the functionality. I'll add more tests in later PRs.

### CODE REVIEW

File: helm/apptest-app/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+appVersion: v1.8.0
+description: A test app for the apptest library
+home: https://github.com/giantswarm/apptest
+name: apptest-app
+version: [[ .Version ]]
```
Comment (rossf7) [inline helm/apptest-app/Chart.yaml:6]: This is a simplified version of https://github.com/giantswarm/test-app

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,3 +12,28 @@ workflows:
...
+      - architect/push-to-app-catalog:
+          context: "architect"
+          name: push-apptest-app-to-control-plane-catalog
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "apptest-app"
+          # Needed to trigger job also on git tag.
+          filters:
+            tags:
+              only: /^v.*/
```
Comment (tomahawk28) [inline .circleci/config.yml:25]: It doesn't need to work on tags. 
```suggestion
              ignore: /^.*/
```

**Review by tomahawk28 — COMMENTED** [review]
I think it should not move into the `control-plane-catalog` in any case. 
Can you block uploading this chart in our official catalog?

File: .circleci/config.yml [hunk 3]
```diff
@@ -12,3 +12,29 @@ workflows:
...
+          context: "architect"
+          name: push-apptest-app-to-control-plane-catalog
+          app_catalog: "control-plane-test-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "apptest-app"
+          filters:
+            # Do not trigger the job on merge to master.
+            branches:
+              ignore:
+                - master
```
Comment (rossf7) [inline .circleci/config.yml:26]: @tomahawk28 I updated the test to only push to the test catalog and skip merges to master. As the chart is only used for the test.

PTAL

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**